In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
import scipy.stats as stats
from scipy.stats import ttest_1samp #One Sample t test
from scipy.stats import ttest_ind #Two Independent Variables t test
from scipy.stats import ttest_rel# Two Dependent Samples Paired t test

One sample T test
It does a Two tailed test by default. That means, the reported P-value will always be computed for a Two-tailed test. To calculate the correct P value, you need to divide the output P-value by 2.
t statistic = Z / s = (x(bar)-Mu)/sigma/n^0.5

In [110]:
x =  [21.5, 24.5, 18.5, 17.2, 14.5, 23.2, 22.1, 20.5, 19.4, 18.1, 24.1, 18.5]

In [111]:
import numpy as np
x=np.array(x)
x

array([21.5, 24.5, 18.5, 17.2, 14.5, 23.2, 22.1, 20.5, 19.4, 18.1, 24.1,
       18.5])

In [112]:
x.mean()

20.175

In [113]:
#Population Mean is assumed =20
tscore, p_value = ttest_1samp(x, popmean=20)
print("t Statistic: ", tscore)  
print("P Value: ", p_value)

t Statistic:  0.2006562773994862
P Value:  0.8446291893053613


In [114]:
if p_value < 0.05:
    print('Reject Null Hypotesis')
else:
    print('Fail to reject Null Hypothesis')

Fail to reject Null Hypothesis


In [7]:
from numpy.random import seed
seed(1)

In [7]:
df=pd.read_excel('Lung_Capacity.xlsx')
df

Age  Height Smoke  Gender  LungCap
0      6    62.1    no    male    6.475
1     18    74.7   yes  female   10.125
2     16    69.7    no  female    9.550
3     14    71.0    no    male   11.125
4      5    56.9    no    male    4.800
..   ...     ...   ...     ...      ...
720    9    56.0    no  female    5.725
721   18    72.0   yes    male    9.050
722   11    60.5   yes  female    3.850
723   15    64.9    no  female    9.825
724   10    67.7    no    male    7.100

[725 rows x 5 columns]

In [94]:
df.describe()

Height     LungCap
count  725.000000  725.000000
mean    64.836276    7.863148
std      7.202144    2.662008
min     45.300000    0.507000
25%     59.900000    6.150000
50%     65.400000    8.000000
75%     70.300000    9.800000
max     81.800000   14.675000

t test - 1 Sample

Default two sided

Ho:Population Age Mean = Sample Mean = 7.86
H1: Population Mean is not equal to Sample Mean
Here Populatiopn mean is a assumed value

T test - 2 Independent Sample

Dependent Variable --> Lung Capacity
Independent Variable --> Smoke
Two Tail test
Assume significance level / alpha = 5%
CI = 95%
Unequal Variance

H0: Mean of Lung Capacity (Y) = Mean of Lung Capacity (N) ie Smoking = No Difference / Impact on Lung Capacity
H1: Mean of Lung Capacity (Y) is not equal to Mean of Lung Capacity (N) 

In [5]:
#Mean
df_smokeyes=df.loc[df['Smoke']=='yes']
df_smokeyes.mean(axis=0)  

Age        14.779221
Height     68.522078
LungCap     8.645455
dtype: float64

In [7]:
#Mean
df_smokeno=df.loc[df['Smoke']=='no']
df_smokeno.mean(axis=0)  

Age        12.035494
Height     64.398302
LungCap     7.770188
dtype: float64

In [10]:
#Variance
df_smokeyes=df.loc[df['Smoke']=='yes']
df_smokeyes.var(axis=0)  

Age         6.937457
Height     27.554638
LungCap     3.545292
dtype: float64

In [11]:
#Variance
df_smokeno=df.loc[df['Smoke']=='no']
df_smokeno.var(axis=0)  

Age        16.331041
Height     52.998467
LungCap     7.431694
dtype: float64

Variance is unequal

In [12]:
#Import label encoder 
from sklearn import preprocessing 
  
# label_encoder object knows how to understand word labels. 
label_encoder = preprocessing.LabelEncoder() 
  
# Encode labels in column 'species'. 
df['Smoke']= label_encoder.fit_transform(df['Smoke']) 
  
df['Smoke'].unique() 

array([0, 1])

In [13]:
print(df['Smoke'])

0      0
1      1
2      0
3      0
4      0
      ..
720    0
721    1
722    1
723    0
724    0
Name: Smoke, Length: 725, dtype: int32


In [14]:
ttest,p_value=stats.ttest_ind(df['LungCap'],df['Smoke'],equal_var=False)

In [15]:
p_value

0.0

In [16]:
if p_value < 0.05:
    print('Reject Null Hypotesis')
else:
    print('Fail to reject Null Hypothesis')

Reject Null Hypotesis


Paired T test

Two Sided
alpha value =5%
p value < 0.05 Reject H0
p value > 0.05 Fail to reject H0

H0:Mean of BP before = Mean of BP after (No Difference) 
H1:Mean of BP before is not equal to Mean of BP after

In [17]:
df1=pd.read_excel('BloodPressure.xls')
df1

Subject  Before  After
0         1     135    127
1         2     142    145
2         3     137    131
3         4     122    125
4         5     147    132
5         6     151    147
6         7     131    119
7         8     117    125
8         9     154    132
9        10     143    139
10       11     133    122
11       12     125    125
12       13     154    138
13       14     156    121
14       15     132    126
15       16     125    127
16       17     136    128
17       18     141    132
18       19     129    131
19       20     148    139
20       21     136    120
21       22     120    119
22       23     147    135
23       24     138    142
24       25     158    130

In [18]:
df1['Before'].mean()

138.28

In [19]:
df1['After'].mean()

130.28

In [20]:
ttest,p_value=stats.ttest_rel(df1['Before'],df1['After'])

In [21]:
p_value

0.0006985894059870637

In [22]:
if p_value < 0.05:
    print('Reject Null Hypotesis')
else:
    print('Fail to reject Null Hypothesis')

Reject Null Hypotesis
